cmd/root.go

```diff
@@ -76,39 +79,65 @@ func New(o *options.Options) *cobra.Command {
 	return cmd
 }
 
-// rootCmd runs scorecard checks given a set of arguments.
-func rootCmd(o *options.Options) error {
-	var err error
-	var repoResult scorecard.Result
+// Build the list of repositories to scan, honoring --repos > --org > --local > --repo/pkg-managers.
+func buildRepoURLs(ctx context.Context, o *options.Options) ([]string, error) {
+	// --repos has highest precedence
+	if len(o.Repos) > 0 {
+		var urls []string
+		for _, r := range o.Repos {
+			r = strings.TrimSpace(r)
+			if r != "" {
+				urls = append(urls, r)
+			}
+		}
+		return urls, nil
+	}
+
+	// --org: expand to all non-archived repos
+	if o.Org != "" {
+		// create a transport to respect auth, rate limiting, etc.
+		logger := sclog.NewLogger(sclog.DefaultLevel)
+		rt := roundtripper.NewTransport(ctx, logger)
+		repos, err := orgpkg.ListOrgRepos(ctx, o.Org, rt)
+		if err != nil {
+			return nil, fmt.Errorf("listing repositories for org %q: %w", o.Org, err)
+		}
+		return repos, nil
+	}
 
+	// --local: single local path
+	if o.Local != "" {
+		return []string{o.Local}, nil
+	}
+
+	// Package managers may override --repo
 	p := &pmc.PackageManagerClient{}
 	// Set `repo` from package managers.
 	pkgResp, err := fetchGitRepositoryFromPackageManagers(o.NPM, o.PyPI, o.RubyGems, o.Nuget, p)
 	if err != nil {
-		return fmt.Errorf("fetchGitRepositoryFromPackageManagers: %w", err)
+		return nil, fmt.Errorf("fetchGitRepositoryFromPackageManagers: %w", err)
 	}
 	if pkgResp.exists {
 		o.Repo = pkgResp.associatedRepo
 	}
 
-	pol, err := policy.ParseFromFile(o.PolicyFile)
-	if err != nil {
-		return fmt.Errorf("readPolicy: %w", err)
-	}
+	return []string{o.Repo}, nil
+}
```

Comment:
just a comment for myself: doing package manager parsing before `--repo` is necessary since as you mention it overwrites `o.Repo`